DependencyInjection/Compiler/DescriptionEnhancerPass.php

```diff
@@ -59,12 +70,26 @@ public function process(ContainerBuilder $container)
             ));
         }
 
+        // Remove inactive enhancers from the container.
+        //
+        // Note that previously we relied on the container automatically removing
+        // unreferenced definitions, but this proved to not always be the case. It
+        // is safer to remove them here.
         $inactiveEnhancers = array_diff($enhancerNames, $enabledEnhancers);
```

Comment:
@WouterJ I tried removing this in this PR, however it caused issues, it would seem that the container does not always remove the unused definitions in some cases (pehaps on initial warmup). Think its safer to do it here.